src/format/mod.rs

```diff
@@ -74,6 +72,18 @@ impl<'a> InputFormatParser<'a> {
             .count()
     }
 
+    #[inline]
+    pub fn get_variable_names(&self) -> Vec<Option<&'a str>> {
+        self.tokens
+            .iter()
+            .filter_map(|token| match token {
+                InputFormatToken::Variable(name) => Some(Some(*name)),
+                InputFormatToken::GenericType => Some(None),
```

Comment:
This must be refactored as we removed support for types it is not now a custom type, is a placeholder without name or anonymous. Think a better name and change it.